grails-app/services/org/pih/warehouse/order/OrderService.groovy

```diff
@@ -222,8 +222,12 @@ class OrderService {
         // update the status of the order before saving
         order.updateStatus()
 
-        order.originParty = order.originParty?:order?.origin?.organization
-        order.destinationParty = order.destinationParty?:order?.destination?.organization
+        if (order.originParty != order?.origin?.organization) {
```

Comment:
Is there need for checking this? If it is already equal, then what's the issue with reassigning it?

Reply:
I though there was, but I think you're right.